File: 1538-maximum-points-you-can-obtain-from-cards/1538-maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& nums, int k) {
         int ans = 0,sum=0,lsum = 0,rsum = 0,ridx = nums.size()-1;
           
            for(int i=0;i<k;i++){
           lsum  += nums[i];
          }
          sum = lsum;
          for(int i = k-1; i>=0;i--){
           
            lsum -= nums[i];
            rsum += nums[ridx--];
           
             sum = max(sum,lsum + rsum);
          }
         
        return sum;
    }
};

File: 1538-maximum-points-you-can-obtain-from-cards/1538-maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] nums, int k) {
        int ans = 0,sum = 0,end = nums.length-1;
        for(int i=0;i<k;i++){
             sum += nums[i];
        }
        ans = Math.max(ans,sum);
        for(int i=k-1;i>=0;i--){
            sum -= nums[i];
            sum += nums[end--];
            ans = Math.max(sum,ans);
        }
        return ans;
    }
}